improve serverless-driver (#16567)
pingcap/docs: tidb-cloud/serverless-driver.md

| `url`        | string   | N/A           | The URL for the database, in the `mysql://[username]:[password]@[host]/[database]` format, where `database` can be skipped if you intend to connect to the default database.                                                                                                                                                                                 |
| `fetch`      | function | global fetch  | Custom fetch function. For example, you can use the `undici` fetch in node.js.                                                                                                                                                                                                                                                                               |
| `arrayMode`  | bool     | `false`       | Whether to return results as arrays instead of objects. To get better performance, set it to `true`.                                                                                                                                                                                                                                                         |

> | `fullResult` | bool     | `false`       | Whether to return full result object instead of just rows. To get more detailed results, set it to `true`.                                                                                                                                                                                                                                                   |

Author: 是的。我用的 idea , formate 了一下表格

| `decoders`   | object | `{}`              | A collection of key-value pairs, which enables you to customize the decoding process for different column types. In each pair, you can specify a column type as the key and specify a corresponding function as the value. This function takes the raw string value received from TiDB Cloud serverless driver as an argument and returns the decoded value. If you have configured `decoders` at both the connection and SQL levels, the key-value pairs with different keys configured at the connection level will be merged to the SQL level to take effect. If the same key (this is, column type) is specified at both levels, the value at the SQL level takes precedence. |

> 

Author: 是的

| `arrayMode`  | bool   | `false`           | Whether to return results as arrays instead of objects. To get better performance, set it to `true`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

> | `fullResult` | bool   | `false`           | Whether to return full result object instead of just rows. To get more detailed results, set it to `true`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| `fullResult` | bool   | `false`           | Whether to return the full result object instead of just rows. To get more detailed results, set it to `true`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
```

| `decoders`   | object | `{}`              | A collection of key-value pairs, which enables you to customize the decoding process for different column types. In each pair, you can specify a column type as the key and specify a corresponding function as the value. This function takes the raw string value received from TiDB Cloud serverless driver as an argument and returns the decoded value. If you have configured `decoders` at both the connection and SQL levels, the key-value pairs with different keys configured at the connection level will be merged to the SQL level to take effect. If the same key (this is, column type) is specified at both levels, the value at the SQL level takes precedence. |

**arrayMode and fullResult**

> To return full result object as arrays, you can configure the `arrayMode` and `fullResult` options as follows:

Reviewer: ```suggestion
To return the full result object as arrays, you can configure the `arrayMode` and `fullResult` options as follows:
```

})
```

- _Changed in v0.0.7_: SQL level option `isolation` is added.

> - _Changed in v0.0.10_: Connection level configuration and SQL level option `decoders` is added.

Reviewer: ```suggestion
> **Note:**
>
> TiDB Cloud serverless driver configuration changes:
> - _v0.0.7_: add the SQL level option`isolation`.
> - _v0.0.10_: add the connection level configuration `decoders` and the SQL level option `decoders`.

})
```

> **Note:**
>
> TiDB Cloud serverless driver configuration changes:
> 

> > - v0.0.7: add the SQL level option`isolation`.

Reviewer 2: ```suggestion
> - v0.0.7: add the SQL level option `isolation`.
```

> > - _v0.0.7_: add the SQL level option`isolation`.
> > - _v0.0.10_: add the connection level configuration `decoders` and the SQL level option `decoders`.

Reviewer: ```suggestion
> - v0.0.7: add the SQL level option`isolation`.
> - v0.0.10: add the connection level configuration `decoders` and the SQL level option `decoders`.
```

| DECIMAL              | string          |
| CHAR                 | string          |
| VARCHAR              | string          |
| BINARY               | Uint8Array      |
| VARBINARY            | Uint8Array      |
| TINYTEXT             | string          |
| TEXT                 | string          |
| MEDIUMTEXT           | string          |
| LONGTEXT             | string          |
| TINYBLOB             | Uint8Array      |
| BLOB                 | Uint8Array      |
| MEDIUMBLOB           | Uint8Array      |
| LONGBLOB             | Uint8Array      |
| DATE                 | string          |
| TIME                 | string          |
| DATETIME             | string          |
| TIMESTAMP            | string          |
| ENUM                 | string          |
| SET                  | string          |
| BIT                  | Uint8Array      |
| JSON                 | object          |
| NULL                 | null            |
| Others               | string          |

> **Note:**
>

> > Please use utf8mb4 collation in TiDB Serverless(default is utf8mb4) for the type convert to Javascript string. Because we will use the `utf8` encoding to decode them to string. 

Reviewer: ```suggestion
> Make sure to use the default `utf8mb4` collation in TiDB Serverless for the type conversion to JavaScript strings, because TiDB Cloud serverless driver uses the `utf8` encoding to decode them to strings. 
```

> 
> - _Changed in v0.1.0_: The `BINARY`、`VARBINARY`、`TINYBLOB`、`BLOB`、`MEDIUMBLOB`、`LONGBLOB` and `BIT` types are now returned as a `Uint8Array` instead of a `string`.

Reviewer: ```suggestion
> **Note:**
>
> TiDB Cloud serverless driver data type mapping changes:
>  _Changed in v0.1.0_: The `BINARY`、`VARBINARY`、`TINYBLOB`、`BLOB`、`MEDIUMBLOB`、`LONGBLOB` and `BIT` types are now returned as a `Uint8Array` instead of a `string`.
```

| CHAR                 | string          |
| VARCHAR              | string          |
| BINARY               | Uint8Array      |
| VARBINARY            | Uint8Array      |
| TINYTEXT             | string          |
| TEXT                 | string          |
| MEDIUMTEXT           | string          |
| LONGTEXT             | string          |
| TINYBLOB             | Uint8Array      |
| BLOB                 | Uint8Array      |
| MEDIUMBLOB           | Uint8Array      |
| LONGBLOB             | Uint8Array      |
| DATE                 | string          |
| TIME                 | string          |
| DATETIME             | string          |
| TIMESTAMP            | string          |
| ENUM                 | string          |
| SET                  | string          |
| BIT                  | Uint8Array      |
| JSON                 | object          |
| NULL                 | null            |
| Others               | string          |

> **Note:**
>
> Make sure to use the default `utf8mb4` collation in TiDB Serverless for the type conversion to JavaScript strings, because TiDB Cloud serverless driver uses the `utf8` encoding to decode them to strings. 

> > **Note:**

Reviewer: ```suggestion

> **Note:**
```

>
> TiDB Cloud serverless driver data type mapping changes:
> 

> > - _v0.1.0_: The `BINARY`、`VARBINARY`、`TINYBLOB`、`BLOB`、`MEDIUMBLOB`、`LONGBLOB` and `BIT` types are now returned as a `Uint8Array` instead of a `string`.

Reviewer: ```suggestion
> - v0.1.0 The `BINARY`、`VARBINARY`、`TINYBLOB`、`BLOB`、`MEDIUMBLOB`、`LONGBLOB` and `BIT` types are now returned as a `Uint8Array` instead of a `string`.
```

| DECIMAL              | string          |
| CHAR                 | string          |
| VARCHAR              | string          |
| BINARY               | Uint8Array      |
| VARBINARY            | Uint8Array      |
| TINYTEXT             | string          |
| TEXT                 | string          |
| MEDIUMTEXT           | string          |
| LONGTEXT             | string          |
| TINYBLOB             | Uint8Array      |
| BLOB                 | Uint8Array      |
| MEDIUMBLOB           | Uint8Array      |
| LONGBLOB             | Uint8Array      |
| DATE                 | string          |
| TIME                 | string          |
| DATETIME             | string          |
| TIMESTAMP            | string          |
| ENUM                 | string          |
| SET                  | string          |
| BIT                  | Uint8Array      |
| JSON                 | object          |
| NULL                 | null            |
| Others               | string          |

> **Note:**
>

> > Make sure to use the default `utf8mb4` charset in TiDB Serverless for the type conversion to JavaScript strings, because TiDB Cloud serverless driver uses the UTF-8 encoding to decode them to strings. 

Reviewer: ```suggestion
> Make sure to use the default `utf8mb4` character set in TiDB Serverless for the type conversion to JavaScript strings, because TiDB Cloud serverless driver uses the UTF-8 encoding to decode them to strings. 
```

> > Make sure to use the default `utf8mb4` collation in TiDB Serverless for the type conversion to JavaScript strings, because TiDB Cloud serverless driver uses the `utf8` encoding to decode them to strings. 

Reviewer 2: ```suggestion
> Make sure to use the default `utf8mb4` charset in TiDB Serverless for the type conversion to JavaScript strings, because TiDB Cloud serverless driver uses the `utf8` encoding to decode them to strings. 
```

> **Note:**
>
> TiDB Cloud serverless driver data type mapping changes:
> 

> > - v0.1.0: the `BINARY`、`VARBINARY`、`TINYBLOB`、`BLOB`、`MEDIUMBLOB`、`LONGBLOB`, and `BIT` types are now returned as a `Uint8Array` instead of a `string`.

Reviewer: ```suggestion
> - v0.1.0: the `BINARY`, `VARBINARY`, `TINYBLOB`, `BLOB`, `MEDIUMBLOB`, `LONGBLOB`, and `BIT` types are now returned as a `Uint8Array` instead of a `string`.
```
